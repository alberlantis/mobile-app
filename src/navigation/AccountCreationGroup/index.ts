export { default } from "./AccountCreationGroup";
export type { AccountCreationParamList } from "./AccountCreationGroup";
