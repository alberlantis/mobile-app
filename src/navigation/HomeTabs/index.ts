export { default } from "./HomeTabs";
export type { HomeTabsParamList } from "./HomeTabs";
