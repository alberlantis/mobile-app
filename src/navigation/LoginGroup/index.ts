export { default } from "./LoginGroup";
export type { LoginGroupParamList } from "./LoginGroup";
