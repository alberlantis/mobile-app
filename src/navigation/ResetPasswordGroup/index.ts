export { default } from "./ResetPasswordGroup";
export type { ResetPasswordGroupParamList } from "./ResetPasswordGroup";
