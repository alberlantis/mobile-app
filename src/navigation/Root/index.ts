export { default } from "./Root";
export type {
  RootScreenProps,
  RootNavigationProps,
  RootRouteProps,
  HomeTabsParams,
} from "./Root";
