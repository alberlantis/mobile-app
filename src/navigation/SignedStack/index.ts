export { default } from "./SignedStack";
export type {
  SignedNavigationProps,
  SignedRouteProps,
  SignedScreenProps,
  SignedParamList,
} from "./SignedStack";
