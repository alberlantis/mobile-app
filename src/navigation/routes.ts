export const SCREENS = {
  HOME_TABS: "HomeTabs",
  HOME: "Home",
  NOTIFICATIONS: "Notifications",
  ONBOARDING: "Onboarding",
  POSTING: "Posting",
  PROFILE_HOME: "ProfileHome",
  SPLASH: "Splash",
  SIGN_UP: "SignUp",
  NOSTR_UP: "NostrUp",
  LOGIN: "Login",
  NOSTR_IN: "NostrIn",
  INTERESTS: "Interests",
  COMPLETE_PROFILE: "CompleteProfile",
  RECOVERY_EMAIL: "RecoveryEmail",
  RESET_PASSWORD: "ResetPassword",
} as const;
