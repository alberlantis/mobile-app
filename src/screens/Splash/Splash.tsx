import React, { useEffect } from "react";
import { Image, SafeAreaView, StatusBar } from "react-native";

import type { RootScreenProps } from "src/navigation/Root";
import { SCREENS } from "src/navigation/routes";
import s from "./Splash.style";

type ISplashProps = RootScreenProps<"Splash">;

const Splash = ({ navigation }: ISplashProps) => {
  useEffect(() => {
    const timer = setTimeout(() => {
      navigation.navigate(SCREENS.ONBOARDING);
    }, 3000);

    return () => {
      clearTimeout(timer);
    };
  }, [navigation]);

  return (
    <SafeAreaView style={s.container}>
      <StatusBar hidden />
      <Image
        source={require("assets/splash/splash.png")}
        style={s.image}
        resizeMode="cover"
        testID="splash-screen-image-id"
      />
    </SafeAreaView>
  );
};

export default Splash;
