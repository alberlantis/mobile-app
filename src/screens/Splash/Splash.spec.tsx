import React from "react";
import { render, screen } from "@testing-library/react-native";

import s from "./Splash.style";
import { SCREENS } from "src/navigation/routes";
import Splash from "./Splash";

jest.useFakeTimers();
jest.mock("src/shared/hooks", () => ({
  useImageAssets: jest.fn(() => ({
    images: {
      splash: {
        testUri: "../../../assets/splash/splash.png",
      },
    },
  })),
}));

const mockNavigation: any = {
  navigate: jest.fn(),
};
const mockRoute: any = {};
const setTimeoutSpy = jest.spyOn(global, "setTimeout");
const clearTimeoutSpy = jest.spyOn(global, "clearTimeout");

describe("Splash", () => {
  beforeEach(() => {
    render(<Splash navigation={mockNavigation} route={mockRoute} />);
  });

  it("should render splash image", () => {
    const image = screen.getByTestId("splash-screen-image-id");
    expect(screen.toJSON()).toMatchSnapshot();
    expect(image).toHaveProp("source", {
      testUri: "../../../assets/splash/splash.png",
    });
    expect(image).toHaveStyle(s.image);
    expect(image).toBeOnTheScreen();
  });

  it("should navigate to Onboarding screen after 3 seconds", () => {
    jest.advanceTimersByTime(3000);
    expect(setTimeoutSpy).toHaveBeenCalledWith(expect.any(Function), 3000);
    expect(clearTimeoutSpy).toHaveBeenCalled();
    expect(mockNavigation.navigate).toHaveBeenCalledTimes(1);
    expect(mockNavigation.navigate).toHaveBeenCalledWith(SCREENS.ONBOARDING);
  });
});
