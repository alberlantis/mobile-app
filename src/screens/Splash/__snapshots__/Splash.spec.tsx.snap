// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash should render splash image 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "black",
      "flex": 1,
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      {
        "testUri": "../../../assets/splash/splash.png",
      }
    }
    style={
      {
        "height": "100%",
        "width": "100%",
      }
    }
    testID="splash-screen-image-id"
  />
</RCTSafeAreaView>
`;
